src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -24,7 +24,10 @@ namespace Microsoft.Diagnostics.Tools.Counters
     public class CounterMonitor
     {
         private const int BufferDelaySecs = 1;
+        private const string SharedSessionId = "SHARED"; // This should be identical to the one used by dotnet-monitor in MetricSourceConfiguration.cs
+        private static HashSet<string> inactiveSharedSessions = new();
```

Comment:
This needs an appropriate StringComparer provided to the constructor, at least to be explicit about the expected comparison type.